src/lib_ccx/utility.c

```diff
@@ -506,36 +506,40 @@ char *get_basename(char *filename)
 	return basefilename;
 }
 
-char *get_file_extension(enum ccx_output_format write_format)
+char *get_file_extension(const enum ccx_output_format write_format)
 {
 	switch (write_format)
 	{
+		case CCX_OF_CCD:
+			return ".ccd";
 		case CCX_OF_RAW:
-			return strdup(".raw");
+			return ".raw";
```

Comment:
You are returning a non-const pointer to data in code. Probably really bad things will happen if the caller tries for example to change the file extension in any way (such as make it uppercase).

https://stackoverflow.com/questions/4836534/returning-a-pointer-to-a-literal-or-constant-character-array-string

Reply:
I'm surprised I haven't done it. Well anyway fixed it.